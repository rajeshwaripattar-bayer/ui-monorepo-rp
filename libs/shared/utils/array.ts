import map from 'lodash/map'
import fill from 'lodash/fill'

export const range = (start: number, stop: number, step: number = 1) =>
  map(fill(Array(Math.ceil((stop - start) / step)), start), (x, y) => x + y * step)
