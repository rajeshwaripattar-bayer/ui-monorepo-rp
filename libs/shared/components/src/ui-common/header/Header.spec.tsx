import { screen } from '@gc/utils'
import { renderWithTestWrapper } from '@gc/shared/test'

import Header from './Header'
import { noop } from 'lodash'

const header = (
  <Header
    title='Test'
    overlineText='myBadge'
    secText1='cool description'
    buttonProps={[
      { label: 'create quote', onClick: noop },
      { label: 'edit', onClick: noop, variant: 'outlined' }
    ]}
  />
)

describe('Header', () => {
  it('should render successfully', () => {
    const { baseElement } = renderWithTestWrapper(
      <Header title='Test Title' secText1='Documentation' overlineText='My Banner' />
    )

    expect(baseElement).toBeTruthy()
    expect(screen.getByText(/Test/)).toBeTruthy()
    expect(screen.getByText(/Documentation/)).toBeTruthy()
    expect(screen.getByText(/My Banner/)).toBeTruthy()
  })

  it('should render successfully with mobile Buttons and no badge text', () => {
    const buttonProps = [
      {
        label: 'more',
        onClick: noop
      },
      { label: 'edit', onClick: noop, variant: 'outlined' },
      {
        label: 'covert to order',
        onClick: noop,
        variant: 'filled'
      }
    ]
    renderWithTestWrapper(<Header title='Test Title' secText1='Documentation' buttonProps={buttonProps} />)
    expect(screen.getByRole('button', { name: 'more' })).toBeDefined()
    expect(screen.getByRole('button', { name: 'covert to order' })).toBeDefined()
  })
  it('should render desktop button if desktop button props were passed', () => {
    renderWithTestWrapper(
      <Header
        title='Test'
        buttonProps={[
          {
            label: 'create quote',
            onClick: noop
          },
          { label: 'edit', onClick: noop, variant: 'outlined' }
        ]}
      />
    )
    expect(screen.getByRole('button', { name: 'create quote' })).toBeDefined()
  })

  it('should render badge, description and title when passed', () => {
    const { getByText } = renderWithTestWrapper(header)
    expect(getByText('cool description')).toBeTruthy()
    expect(getByText('myBadge')).toBeTruthy()
  })

  it('Should render component for screen width greater than 1440', () => {
    const wrapper = renderWithTestWrapper(header, { width: 1440 })
    expect(wrapper).toMatchSnapshot()
  })

  it('Should render component for screen width smaller than 1440', () => {
    const wrapper = renderWithTestWrapper(header, { width: 1439 })
    expect(wrapper).toMatchSnapshot()
  })

  it('Should render component for screen width smaller than 1024', () => {
    const wrapper = renderWithTestWrapper(header, { width: 1023 })
    expect(wrapper).toMatchSnapshot()
  })

  it('Should render component for screen width smaller than 840', () => {
    const wrapper = renderWithTestWrapper(header, { width: 839 })
    expect(wrapper).toMatchSnapshot()
  })

  it('Should render component for screen width smaller than 720', () => {
    const wrapper = renderWithTestWrapper(header, { width: 719 })
    expect(wrapper).toMatchSnapshot()
  })

  it('Should render component for screen width smaller than 600', () => {
    const wrapper = renderWithTestWrapper(header, { width: 599 })
    expect(wrapper).toMatchSnapshot()
  })
})
