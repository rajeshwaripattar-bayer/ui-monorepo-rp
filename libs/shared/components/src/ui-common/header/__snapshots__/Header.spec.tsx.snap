// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should render component for screen width greater than 1440 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="header_main"
        >
          <div
            class="lmnt lmnt-badge lmnt-badge--label lmnt-badge--label-gray mdc-typography--caption-bold"
          >
            <span
              class="lmnt-badge--label__label"
            >
              myBadge
            </span>
          </div>
          <span
            class="mdc-typography lmnt mdc-typography--headline3"
          >
            Test
          </span>
          <span
            class="mdc-typography lmnt mdc-typography--body2"
          >
            cool description
          </span>
        </div>
        <div
          class="desktopButtonGroup"
        >
          <div
            class="lmnt lmnt-group"
            style="display: flex; flex-direction: row; justify-content: flex-start; align-items: baseline; gap: 8px;"
          >
            <button
              class="lmnt mdc-button lmnt-button mdc-button--touch lmnt-button--filled-primary mdc-button--raised lmnt-button--themeDefault mdc-touch-target-wrapper"
              type="button"
            >
              <span
                class="mdc-button__ripple"
              />
              <span
                class="mdc-button__label"
              >
                create quote
              </span>
              <span
                class="mdc-button__touch"
              />
            </button>
            <button
              class="lmnt mdc-button lmnt-button mdc-button--touch lmnt-button--outlined-primary mdc-button--outlined lmnt-button--themeDefault mdc-touch-target-wrapper"
              type="button"
            >
              <span
                class="mdc-button__ripple"
              />
              <span
                class="mdc-button__label"
              >
                edit
              </span>
              <span
                class="mdc-button__touch"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="header_main"
      >
        <div
          class="lmnt lmnt-badge lmnt-badge--label lmnt-badge--label-gray mdc-typography--caption-bold"
        >
          <span
            class="lmnt-badge--label__label"
          >
            myBadge
          </span>
        </div>
        <span
          class="mdc-typography lmnt mdc-typography--headline3"
        >
          Test
        </span>
        <span
          class="mdc-typography lmnt mdc-typography--body2"
        >
          cool description
        </span>
      </div>
      <div
        class="desktopButtonGroup"
      >
        <div
          class="lmnt lmnt-group"
          style="display: flex; flex-direction: row; justify-content: flex-start; align-items: baseline; gap: 8px;"
        >
          <button
            class="lmnt mdc-button lmnt-button mdc-button--touch lmnt-button--filled-primary mdc-button--raised lmnt-button--themeDefault mdc-touch-target-wrapper"
            type="button"
          >
            <span
              class="mdc-button__ripple"
            />
            <span
              class="mdc-button__label"
            >
              create quote
            </span>
            <span
              class="mdc-button__touch"
            />
          </button>
          <button
            class="lmnt mdc-button lmnt-button mdc-button--touch lmnt-button--outlined-primary mdc-button--outlined lmnt-button--themeDefault mdc-touch-target-wrapper"
            type="button"
          >
            <span
              class="mdc-button__ripple"
            />
            <span
              class="mdc-button__label"
            >
              edit
            </span>
            <span
              class="mdc-button__touch"
            />
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": undefined,
  "unmount": [Function],
}
`;

exports[`Header Should render component for screen width smaller than 600 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="header_main"
        >
          <div
            class="lmnt lmnt-badge lmnt-badge--label lmnt-badge--label-gray mdc-typography--caption-bold"
          >
            <span
              class="lmnt-badge--label__label"
            >
              myBadge
            </span>
          </div>
          <span
            class="mdc-typography lmnt mdc-typography--headline5"
          >
            Test
          </span>
          <span
            class="mdc-typography lmnt mdc-typography--body2"
          >
            <span
              class="mdc-typography lmnt mdc-typography--caption"
            >
               
              cool description
            </span>
          </span>
          <span
            class="mdc-typography lmnt mdc-typography--body2"
          />
        </div>
        <div
          class="lmnt lmnt-group buttonGroup"
          style="display: flex; flex-direction: column; justify-content: flex-start; align-items: baseline; gap: 8px;"
        >
          <button
            class="lmnt mdc-button lmnt-button mdc-button--touch lmnt-button--filled-primary mdc-button--raised lmnt-button--full-width lmnt-button--themeDefault mdc-touch-target-wrapper"
            type="button"
          >
            <span
              class="mdc-button__ripple"
            />
            <span
              class="mdc-button__label"
            >
              create quote
            </span>
            <span
              class="mdc-button__touch"
            />
          </button>
          <button
            class="lmnt mdc-button lmnt-button mdc-button--touch lmnt-button--outlined-primary mdc-button--outlined lmnt-button--full-width lmnt-button--themeDefault mdc-touch-target-wrapper"
            type="button"
          >
            <span
              class="mdc-button__ripple"
            />
            <span
              class="mdc-button__label"
            >
              edit
            </span>
            <span
              class="mdc-button__touch"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="header_main"
      >
        <div
          class="lmnt lmnt-badge lmnt-badge--label lmnt-badge--label-gray mdc-typography--caption-bold"
        >
          <span
            class="lmnt-badge--label__label"
          >
            myBadge
          </span>
        </div>
        <span
          class="mdc-typography lmnt mdc-typography--headline5"
        >
          Test
        </span>
        <span
          class="mdc-typography lmnt mdc-typography--body2"
        >
          <span
            class="mdc-typography lmnt mdc-typography--caption"
          >
             
            cool description
          </span>
        </span>
        <span
          class="mdc-typography lmnt mdc-typography--body2"
        />
      </div>
      <div
        class="lmnt lmnt-group buttonGroup"
        style="display: flex; flex-direction: column; justify-content: flex-start; align-items: baseline; gap: 8px;"
      >
        <button
          class="lmnt mdc-button lmnt-button mdc-button--touch lmnt-button--filled-primary mdc-button--raised lmnt-button--full-width lmnt-button--themeDefault mdc-touch-target-wrapper"
          type="button"
        >
          <span
            class="mdc-button__ripple"
          />
          <span
            class="mdc-button__label"
          >
            create quote
          </span>
          <span
            class="mdc-button__touch"
          />
        </button>
        <button
          class="lmnt mdc-button lmnt-button mdc-button--touch lmnt-button--outlined-primary mdc-button--outlined lmnt-button--full-width lmnt-button--themeDefault mdc-touch-target-wrapper"
          type="button"
        >
          <span
            class="mdc-button__ripple"
          />
          <span
            class="mdc-button__label"
          >
            edit
          </span>
          <span
            class="mdc-button__touch"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": undefined,
  "unmount": [Function],
}
`;

exports[`Header Should render component for screen width smaller than 720 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="header_main"
        >
          <div
            class="lmnt lmnt-badge lmnt-badge--label lmnt-badge--label-gray mdc-typography--caption-bold"
          >
            <span
              class="lmnt-badge--label__label"
            >
              myBadge
            </span>
          </div>
          <span
            class="mdc-typography lmnt mdc-typography--headline5"
          >
            Test
          </span>
          <span
            class="mdc-typography lmnt mdc-typography--body2"
          >
            <span
              class="mdc-typography lmnt mdc-typography--caption"
            >
               
              cool description
            </span>
          </span>
          <span
            class="mdc-typography lmnt mdc-typography--body2"
          />
        </div>
        <div
          class="lmnt lmnt-group buttonGroup"
          style="display: flex; flex-direction: column; justify-content: flex-start; align-items: baseline; gap: 8px;"
        >
          <button
            class="lmnt mdc-button lmnt-button mdc-button--touch lmnt-button--filled-primary mdc-button--raised lmnt-button--full-width lmnt-button--themeDefault mdc-touch-target-wrapper"
            type="button"
          >
            <span
              class="mdc-button__ripple"
            />
            <span
              class="mdc-button__label"
            >
              create quote
            </span>
            <span
              class="mdc-button__touch"
            />
          </button>
          <button
            class="lmnt mdc-button lmnt-button mdc-button--touch lmnt-button--outlined-primary mdc-button--outlined lmnt-button--full-width lmnt-button--themeDefault mdc-touch-target-wrapper"
            type="button"
          >
            <span
              class="mdc-button__ripple"
            />
            <span
              class="mdc-button__label"
            >
              edit
            </span>
            <span
              class="mdc-button__touch"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="header_main"
      >
        <div
          class="lmnt lmnt-badge lmnt-badge--label lmnt-badge--label-gray mdc-typography--caption-bold"
        >
          <span
            class="lmnt-badge--label__label"
          >
            myBadge
          </span>
        </div>
        <span
          class="mdc-typography lmnt mdc-typography--headline5"
        >
          Test
        </span>
        <span
          class="mdc-typography lmnt mdc-typography--body2"
        >
          <span
            class="mdc-typography lmnt mdc-typography--caption"
          >
             
            cool description
          </span>
        </span>
        <span
          class="mdc-typography lmnt mdc-typography--body2"
        />
      </div>
      <div
        class="lmnt lmnt-group buttonGroup"
        style="display: flex; flex-direction: column; justify-content: flex-start; align-items: baseline; gap: 8px;"
      >
        <button
          class="lmnt mdc-button lmnt-button mdc-button--touch lmnt-button--filled-primary mdc-button--raised lmnt-button--full-width lmnt-button--themeDefault mdc-touch-target-wrapper"
          type="button"
        >
          <span
            class="mdc-button__ripple"
          />
          <span
            class="mdc-button__label"
          >
            create quote
          </span>
          <span
            class="mdc-button__touch"
          />
        </button>
        <button
          class="lmnt mdc-button lmnt-button mdc-button--touch lmnt-button--outlined-primary mdc-button--outlined lmnt-button--full-width lmnt-button--themeDefault mdc-touch-target-wrapper"
          type="button"
        >
          <span
            class="mdc-button__ripple"
          />
          <span
            class="mdc-button__label"
          >
            edit
          </span>
          <span
            class="mdc-button__touch"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": undefined,
  "unmount": [Function],
}
`;

exports[`Header Should render component for screen width smaller than 840 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="header_main"
        >
          <div
            class="lmnt lmnt-badge lmnt-badge--label lmnt-badge--label-gray mdc-typography--caption-bold"
          >
            <span
              class="lmnt-badge--label__label"
            >
              myBadge
            </span>
          </div>
          <span
            class="mdc-typography lmnt mdc-typography--headline5"
          >
            Test
          </span>
          <span
            class="mdc-typography lmnt mdc-typography--body2"
          >
            cool description
          </span>
        </div>
        <div
          class="lmnt lmnt-group buttonGroup"
          style="display: flex; flex-direction: row; justify-content: flex-start; align-items: baseline; gap: 8px;"
        >
          <button
            class="lmnt mdc-button lmnt-button mdc-button--touch lmnt-button--filled-primary mdc-button--raised lmnt-button--themeDefault mdc-touch-target-wrapper"
            type="button"
          >
            <span
              class="mdc-button__ripple"
            />
            <span
              class="mdc-button__label"
            >
              create quote
            </span>
            <span
              class="mdc-button__touch"
            />
          </button>
          <button
            class="lmnt mdc-button lmnt-button mdc-button--touch lmnt-button--outlined-primary mdc-button--outlined lmnt-button--themeDefault mdc-touch-target-wrapper"
            type="button"
          >
            <span
              class="mdc-button__ripple"
            />
            <span
              class="mdc-button__label"
            >
              edit
            </span>
            <span
              class="mdc-button__touch"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="header_main"
      >
        <div
          class="lmnt lmnt-badge lmnt-badge--label lmnt-badge--label-gray mdc-typography--caption-bold"
        >
          <span
            class="lmnt-badge--label__label"
          >
            myBadge
          </span>
        </div>
        <span
          class="mdc-typography lmnt mdc-typography--headline5"
        >
          Test
        </span>
        <span
          class="mdc-typography lmnt mdc-typography--body2"
        >
          cool description
        </span>
      </div>
      <div
        class="lmnt lmnt-group buttonGroup"
        style="display: flex; flex-direction: row; justify-content: flex-start; align-items: baseline; gap: 8px;"
      >
        <button
          class="lmnt mdc-button lmnt-button mdc-button--touch lmnt-button--filled-primary mdc-button--raised lmnt-button--themeDefault mdc-touch-target-wrapper"
          type="button"
        >
          <span
            class="mdc-button__ripple"
          />
          <span
            class="mdc-button__label"
          >
            create quote
          </span>
          <span
            class="mdc-button__touch"
          />
        </button>
        <button
          class="lmnt mdc-button lmnt-button mdc-button--touch lmnt-button--outlined-primary mdc-button--outlined lmnt-button--themeDefault mdc-touch-target-wrapper"
          type="button"
        >
          <span
            class="mdc-button__ripple"
          />
          <span
            class="mdc-button__label"
          >
            edit
          </span>
          <span
            class="mdc-button__touch"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": undefined,
  "unmount": [Function],
}
`;

exports[`Header Should render component for screen width smaller than 1024 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="header_main"
        >
          <div
            class="lmnt lmnt-badge lmnt-badge--label lmnt-badge--label-gray mdc-typography--caption-bold"
          >
            <span
              class="lmnt-badge--label__label"
            >
              myBadge
            </span>
          </div>
          <span
            class="mdc-typography lmnt mdc-typography--headline4"
          >
            Test
          </span>
          <span
            class="mdc-typography lmnt mdc-typography--body2"
          >
            cool description
          </span>
        </div>
        <div
          class="lmnt lmnt-group buttonGroup"
          style="display: flex; flex-direction: row; justify-content: flex-start; align-items: baseline; gap: 8px;"
        >
          <button
            class="lmnt mdc-button lmnt-button mdc-button--touch lmnt-button--filled-primary mdc-button--raised lmnt-button--themeDefault mdc-touch-target-wrapper"
            type="button"
          >
            <span
              class="mdc-button__ripple"
            />
            <span
              class="mdc-button__label"
            >
              create quote
            </span>
            <span
              class="mdc-button__touch"
            />
          </button>
          <button
            class="lmnt mdc-button lmnt-button mdc-button--touch lmnt-button--outlined-primary mdc-button--outlined lmnt-button--themeDefault mdc-touch-target-wrapper"
            type="button"
          >
            <span
              class="mdc-button__ripple"
            />
            <span
              class="mdc-button__label"
            >
              edit
            </span>
            <span
              class="mdc-button__touch"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="header_main"
      >
        <div
          class="lmnt lmnt-badge lmnt-badge--label lmnt-badge--label-gray mdc-typography--caption-bold"
        >
          <span
            class="lmnt-badge--label__label"
          >
            myBadge
          </span>
        </div>
        <span
          class="mdc-typography lmnt mdc-typography--headline4"
        >
          Test
        </span>
        <span
          class="mdc-typography lmnt mdc-typography--body2"
        >
          cool description
        </span>
      </div>
      <div
        class="lmnt lmnt-group buttonGroup"
        style="display: flex; flex-direction: row; justify-content: flex-start; align-items: baseline; gap: 8px;"
      >
        <button
          class="lmnt mdc-button lmnt-button mdc-button--touch lmnt-button--filled-primary mdc-button--raised lmnt-button--themeDefault mdc-touch-target-wrapper"
          type="button"
        >
          <span
            class="mdc-button__ripple"
          />
          <span
            class="mdc-button__label"
          >
            create quote
          </span>
          <span
            class="mdc-button__touch"
          />
        </button>
        <button
          class="lmnt mdc-button lmnt-button mdc-button--touch lmnt-button--outlined-primary mdc-button--outlined lmnt-button--themeDefault mdc-touch-target-wrapper"
          type="button"
        >
          <span
            class="mdc-button__ripple"
          />
          <span
            class="mdc-button__label"
          >
            edit
          </span>
          <span
            class="mdc-button__touch"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": undefined,
  "unmount": [Function],
}
`;

exports[`Header Should render component for screen width smaller than 1440 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="header_main"
        >
          <div
            class="lmnt lmnt-badge lmnt-badge--label lmnt-badge--label-gray mdc-typography--caption-bold"
          >
            <span
              class="lmnt-badge--label__label"
            >
              myBadge
            </span>
          </div>
          <span
            class="mdc-typography lmnt mdc-typography--headline4"
          >
            Test
          </span>
          <span
            class="mdc-typography lmnt mdc-typography--body2"
          >
            cool description
          </span>
        </div>
        <div
          class="desktopButtonGroup"
        >
          <div
            class="lmnt lmnt-group"
            style="display: flex; flex-direction: row; justify-content: flex-start; align-items: baseline; gap: 8px;"
          >
            <button
              class="lmnt mdc-button lmnt-button mdc-button--touch lmnt-button--filled-primary mdc-button--raised lmnt-button--themeDefault mdc-touch-target-wrapper"
              type="button"
            >
              <span
                class="mdc-button__ripple"
              />
              <span
                class="mdc-button__label"
              >
                create quote
              </span>
              <span
                class="mdc-button__touch"
              />
            </button>
            <button
              class="lmnt mdc-button lmnt-button mdc-button--touch lmnt-button--outlined-primary mdc-button--outlined lmnt-button--themeDefault mdc-touch-target-wrapper"
              type="button"
            >
              <span
                class="mdc-button__ripple"
              />
              <span
                class="mdc-button__label"
              >
                edit
              </span>
              <span
                class="mdc-button__touch"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="header_main"
      >
        <div
          class="lmnt lmnt-badge lmnt-badge--label lmnt-badge--label-gray mdc-typography--caption-bold"
        >
          <span
            class="lmnt-badge--label__label"
          >
            myBadge
          </span>
        </div>
        <span
          class="mdc-typography lmnt mdc-typography--headline4"
        >
          Test
        </span>
        <span
          class="mdc-typography lmnt mdc-typography--body2"
        >
          cool description
        </span>
      </div>
      <div
        class="desktopButtonGroup"
      >
        <div
          class="lmnt lmnt-group"
          style="display: flex; flex-direction: row; justify-content: flex-start; align-items: baseline; gap: 8px;"
        >
          <button
            class="lmnt mdc-button lmnt-button mdc-button--touch lmnt-button--filled-primary mdc-button--raised lmnt-button--themeDefault mdc-touch-target-wrapper"
            type="button"
          >
            <span
              class="mdc-button__ripple"
            />
            <span
              class="mdc-button__label"
            >
              create quote
            </span>
            <span
              class="mdc-button__touch"
            />
          </button>
          <button
            class="lmnt mdc-button lmnt-button mdc-button--touch lmnt-button--outlined-primary mdc-button--outlined lmnt-button--themeDefault mdc-touch-target-wrapper"
            type="button"
          >
            <span
              class="mdc-button__ripple"
            />
            <span
              class="mdc-button__label"
            >
              edit
            </span>
            <span
              class="mdc-button__touch"
            />
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": undefined,
  "unmount": [Function],
}
`;
